docs/architecture/adr-043-nft-module.md

```diff
@@ -35,341 +36,224 @@ The current design is based on the work done by [IRISnet team](https://github.co
-  m.keeper.bank.BurnCoins(ctx, types.ModuleName, coins)
+  // change ownership mapping
+  m.k.SetNftOwner(msg.Genre, msg.Id, msg.Receiver)
 
-  // Delete bank.Metadata (keeper method not available)
-  
-  nftStoreByType := m.keeper.getNFTStoreByType(ctx, msg.NFT.Type)
-  nftStoreByType.Delete(msg.Id)
-  
-  nftStore := m.keeper.getNFTStore(ctx)
-  nftStore.Delete(baseDenom)
-
-  return &types.MsgBurnResponse{}, nil
+  return &types.MsgSendResponse{}, nil
 }
-```
-
-The upper application calls those methods by holding the MsgClient instance of the `x/nft` module. The execution authority of msg is guaranteed by the OCAPs mechanism.
 
 The query service methods for the `x/nft` module are:
 
 ```proto
 service Query {
 
-  // NFT queries NFT details based on id.
-  rpc NFT(QueryNFTRequest) returns (QueryNFTResponse) {
-    option (google.api.http).get = "/cosmos/nft/v1beta1/nfts/{id}";
+  // Balance queries the number of NFTs based on the genre and owner, same as balanceOf in ERC721
+  rpc Balance(QueryBalanceRequest) returns (QueryBalanceResponse) {
+    option (google.api.http).get = "/cosmos/nft/v1beta1/balance/{genre}/{owner}";
   }
 
-  // NFTs queries all NFTs based on the optional owner.
-  rpc NFTs(QueryNFTsRequest) returns (QueryNFTsResponse) {
-    option (google.api.http).get = "/cosmos/nft/v1beta1/nfts";
+  // Owner queries the owner of the NFT based on the genre and id, same as ownerOf in ERC721
+  rpc Owner(QueryOwnerRequest) returns (QueryOwnerResponse) {
+    option (google.api.http).get = "/cosmos/nft/v1beta1/owner/{genre}/{id}";
   }
 
-  // NFTsOf queries all NFTs based on the type.
-  rpc NFTsOf(QueryNFTsOfRequest) returns (QueryNFTsOfResponse) {
-    option (google.api.http).get = "/cosmos/nft/v1beta1/nfts/{type}";
+  // Supply queries the number of NFTs based on the genre, same as totalSupply in ERC721Enumerable
+  rpc Supply(QuerySupplyRequest) returns (QuerySupplyResponse) {
+    option (google.api.http).get = "/cosmos/nft/v1beta1/supply/{genre}";
   }
 
-  // Supply queries the number of nft based on the type, same as totalSupply of ERC721
-  rpc Supply(QuerySupplyRequest) returns (QuerySupplyResponse) {
-    option (google.api.http).get = "/cosmos/nft/v1beta1/supply/{type}";
+  // NFTsOf queries all NFTs based on the genre, similar to tokenByIndex in ERC721Enumerable
+  rpc NFTs(QueryNFTsRequest) returns (QueryNFTsResponse) {
+    option (google.api.http).get = "/cosmos/nft/v1beta1/nfts/{genre}";
   }
 
-  // Balance queries the number of NFTs based on the owner and type, same as balanceOf of ERC721
-  rpc Balance(QueryBalanceRequest) returns (QueryBalanceResponse) {
-    option (google.api.http).get = "/cosmos/nft/v1beta1/balance/{owner}/{type}";
+  // NFTsOfOwner queries the NFTs based on the genre and owner, similar to tokenOfOwnerByIndex in ERC721Enumerable
+  rpc NFTsOfOwner(QueryNFTsOfOwnerRequest) returns (QueryNFTsResponse) {
+    option (google.api.http).get = "/cosmos/nft/v1beta1/balance/{genre}/{owner}";
```

Comment:
doesn't this route need to be distinguished from the same format used in `Owner(QueryOwnerRequest)`?